File: internal/handlers/importer.go
package handlers

import (
	"github.com/davidmovas/postulator/internal/dto"
	"github.com/davidmovas/postulator/internal/infra/importer"
	"github.com/davidmovas/postulator/pkg/ctx"
)

type ImporterHandler struct {
	service importer.Service
}

func NewImporterHandler(service importer.Service) *ImporterHandler {
	return &ImporterHandler{service: service}
}

func (h *ImporterHandler) ImportTopics(filePath string) *dto.Response[*dto.ImportResult] {
	res, err := h.service.ImportTopics(ctx.FastCtx(), filePath)
	if err != nil {
		return fail[*dto.ImportResult](err)
	}

	return ok(dto.NewImportResult(res))
}

func (h *ImporterHandler) ImportAndAssignToSite(filePath string, siteID int64) *dto.Response[*dto.ImportResult] {
	res, err := h.service.ImportAndAssignToSite(ctx.FastCtx(), filePath, siteID)
	if err != nil {
		return fail[*dto.ImportResult](err)
	}

	return ok(dto.NewImportResult(res))
}

File: frontend/src/services/importer.ts
import { dto } from "@/wailsjs/wailsjs/go/models";
import { ImportAndAssignToSite, ImportTopics } from "@/wailsjs/wailsjs/go/handlers/ImporterHandler";
import { unwrapResponse } from "@/lib/api-utils";
import { ImportResult, mapImportResult } from "@/models/importer";

export const importerService = {
    async importTopics(filePath: string): Promise<ImportResult> {
        const response = await ImportTopics(filePath);
        const result = unwrapResponse<dto.ImportResult>(response);
        return mapImportResult(result);
    },

    async importAndAssignToSite(filePath: string, siteId: number): Promise<ImportResult> {
        const response = await ImportAndAssignToSite(filePath, siteId);
        const result = unwrapResponse<dto.ImportResult>(response);
        return mapImportResult(result);
    }
};